packages/debug/src/browser/debug-session-connection.ts

```diff
@@ -186,44 +194,53 @@ export class DebugSessionConnection implements Disposable {
     sendCustomRequest<T extends DebugProtocol.Response>(command: string, args?: any): Promise<T> {
         return this.doSendRequest<T>(command, args);
     }
-    protected async doSendRequest<K extends DebugProtocol.Response>(command: string, args?: any): Promise<K> {
+
+    protected cancelPendingRequests(): void {
+        this.pendingRequests.forEach((deferred, requestId) => {
```

Comment:
Hmm, forEach is meant to be faster, but I tend to avoid it when dealing with promises.

Reply:
IMO, both versions are perfectly acceptable 🤷

Reply:
@thegecko faster on what front?

Reply:
@tsmaeder right, you may take this comment with a grain of salt as I am nitpicking. But I don't like how the `forEach` method gives us elements in (value, key) order rather than (key, value) order, and also I don't understand why we would want to define anonymous functions to process elements when the language itself gives us `for` loops from the get-go. Feel free to leave as is.

Reply:
Following @thegecko comment I just tried comparing `for of` vs `.forEach` for maps, and `.forEach` seems to win everytime, call me enlightened... Why are we using TS/JS again? /jk

With that in mind, please keep the code as is.